src/NServiceBus.Core/Recoverability/RecoverabilityAction.cs

```diff
@@ -30,6 +30,8 @@ public static RecoverabilityAction ImmediateRetry()
         /// <returns>Delayed retry action.</returns>
         public static RecoverabilityAction DelayedRetry(TimeSpan timeSpan)
         {
+            Guard.AgainstNegativeAndZero(nameof(timeSpan), timeSpan);
```

Comment:
Zero should be fine. I would currently check only against Negative. But I have to dig deeper in Delayed Retries